packages/flutter/test/material/segmented_button_test.dart

```diff
@@ -774,8 +774,48 @@ testWidgets('SegmentedButton shows checkboxes for selected segments', (WidgetTes
     await tester.pumpAndSettle();
     expect(overlayColor(), paints..rect(color: foregroundColor.withOpacity(0.08)));
   });
+
+  testWidgets('Disabled SegmentedButton has correct states when rebuilding', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: Center(
+            child: StatefulBuilder(
+                builder: (BuildContext context, StateSetter setState) {
+                  return Column(
+                    children: <Widget>[
+                      SegmentedButton<int>(
+                        segments: const <ButtonSegment<int>>[
+                          ButtonSegment<int>(value: 0, label: Text('foo')),
+                        ],
+                        selected: const <int>{0},
+                      ),
+                      ElevatedButton(
+                        onPressed: () => setState(() {}),
+                        child: const Text('Trigger rebuild'),
+                      ),
+                    ],
+                  );
+                }
+            ),
+          ),
+        ),
+      ),
+    );
+    final Set<MaterialState> states = <MaterialState>{ MaterialState.selected, MaterialState.disabled };
+    // Check the initial states.
+    SegmentedButtonState<int> state = tester.state(find.byType(SegmentedButton<int>));
+    expect(state.statesControllers.values.first.value, states);
+    // Trigger a rebuild.
+    await tester.tap(find.byType(ElevatedButton));
+    await tester.pumpAndSettle();
+    // Check the states after the rebuild.
+    state = tester.state(find.byType(SegmentedButton<int>));
+    expect(state.statesControllers.values.first.value, states);
+  });
 }
 
 Set<MaterialState> enabled = const <MaterialState>{};
 Set<MaterialState> disabled = const <MaterialState>{ MaterialState.disabled };
 Set<MaterialState> selected = const <MaterialState>{ MaterialState.selected };
+
```

Comment:
```suggestion
```

Reply:
https://dart.dev/tools/linter-rules/eol_at_end_of_file (sure?) I can remove it if it is okay but against linter

Reply:
I might be mistaking.  This is the error from the analyzer in the PR

`║    info • Missing a newline at the end of the file • packages/flutter/test/material/segmented_button_test.dart:820:79 • eol_at_end_of_file`

Reply:
I was), fixed